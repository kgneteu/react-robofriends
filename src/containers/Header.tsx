import React, {SyntheticEvent, useEffect, useState} from "react";
import SearchBox from "../components/SearchBox";
import * as PropTypes from "prop-types";

type ComponentProps = {
    searchChange(event: SyntheticEvent<HTMLInputElement>): void
}
const Header = ({searchChange}: ComponentProps) => {
    const [scrolled, setScrolled] = useState(false);

    useEffect(() => {
        const handleScroll = () => {
            setScrolled(window.scrollY > 8);
        }
        window.addEventListener("scroll", handleScroll)
        return () => {
            window.removeEventListener("scroll", () => handleScroll)
        }
    }, [])

    // console.log('repaint')
    return (
        <header
            style={{
                position: "fixed",
                top: 0,
                textAlign: "center",
                width: "100%",
                height: "200px",
                zIndex: 100,
                boxShadow: scrolled ? "0 4px 4px rgba(0, 0, 0, 0.4)" : "none",
                transition: "all 0.2s ease-out",
            }}>
            <h1 className="f1 center">RoboFriends</h1>
            <SearchBox className={"ml-auto"} searchChange={searchChange}/>
        </header>
    )
};


Header.propTypes = {
    searchChange: PropTypes.func
};

export default React.memo(Header);
