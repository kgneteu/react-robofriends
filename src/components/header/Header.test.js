import {shallow} from "enzyme";
import Header from "./Header";
import {create} from "react-test-renderer";


describe('Header', function () {
    it('should render Header component', function () {
        expect(shallow(<Header/>)).toMatchSnapshot()
    });
    //React Test Renderer
    it('should render Header component alt', function () {
        const header = create(<Header/>)
        expect(header.toJSON()).toMatchSnapshot();
    });
});
