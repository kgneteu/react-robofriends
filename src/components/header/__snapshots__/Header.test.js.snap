// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render Header component 1`] = `
<header
  style={
    Object {
      "boxShadow": "none",
      "height": "200px",
      "position": "fixed",
      "textAlign": "center",
      "top": 0,
      "transition": "all 0.2s ease-out",
      "width": "100%",
      "zIndex": 100,
    }
  }
>
  <h1
    className="f1 center"
  >
    RoboFriends
  </h1>
  <SearchBox
    className="ml-auto"
  />
</header>
`;

exports[`Header should render Header component alt 1`] = `
<header
  style={
    Object {
      "boxShadow": "none",
      "height": "200px",
      "position": "fixed",
      "textAlign": "center",
      "top": 0,
      "transition": "all 0.2s ease-out",
      "width": "100%",
      "zIndex": 100,
    }
  }
>
  <h1
    className="f1 center"
  >
    RoboFriends
  </h1>
  <div
    className="pa2"
  >
    <input
      className="pa3 ba b--green bg-lightest-blue"
      placeholder="search robots"
      type="search"
    />
  </div>
</header>
`;
